### ISSUE TITLE
Cant get zeromq to work on Mac M1

### ISSUE THREAD

**Issue by RubensGitHub** [zeromq/zeromq.js#477]
Disclaimer: Im new to the Mac world and not too familiar with zmq. I would still appreciate help here :) 
On my Macbook M1, I am not getting zmq to work. While the installation throws no errors (`yarn add zeromq`, `yarn add zeromq@6.0.0-beta.6`, also `yarn add zeromq@6.0.0-beta.6 --build-from-source`), it throws symbol errors when I start the application in which zeromq is used:

```
dyld: lazy symbol binding failed: Symbol not found: _zmq_ctx_new
  Referenced from: /Users/ruben/Work/Chrysalis_Ports/industry-marketplace/ServiceApp/server/node_modules/zeromq/build/Release/zeromq.node
  Expected in: flat namespace

dyld: Symbol not found: _zmq_ctx_new
  Referenced from: /Users/ruben/Work/Chrysalis_Ports/industry-marketplace/ServiceApp/server/node_modules/zeromq/build/Release/zeromq.node
  Expected in: flat namespace
```

Im doing everything in a Rosetta-Terminal btw.
I did some googling and tried `npm set config arch x64` and `npm config set platform Mac` and rebuilt the packages, then Im getting:

```
internal/modules/cjs/loader.js:1131
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: dlopen(/Users/ruben/Work/Chrysalis_Ports/industry-marketplace/ServiceApp/server/node_modules/zeromq/build/Release/zeromq.node, 1): no suitable image found.  Did find:
	/Users/ruben/Work/Chrysalis_Ports/industry-marketplace/ServiceApp/server/node_modules/zeromq/build/Release/zeromq.node: mach-o, but wrong architecture
	/Users/ruben/Work/Chrysalis_Ports/industry-marketplace/ServiceApp/server/node_modules/zeromq/build/Release/zeromq.node: mach-o, but wrong architecture

```

**Reproducing**
Apple M1 Chip, Rosetta Terminal -> add latest zeromq version via yarn in existing application, start application.

**Expected behavior**
Smooth application start

**Tested on**
 - Macbook M1
 - OS: MACOS 11.2.3
 - ZeroMQ.js version: zeromq@6.0.0-beta.6 
 - node version 14.17.0

**Comment by alan-ws** [comment]
Would appreciate some help with this

**Comment by Y-qwq** [comment]
try reinstall Node.js with an arm64 arch

**Comment by belloyang** [comment]
I think you will need to build zeromq for arm64 architecture.  The current source code doesn't support building for arm64 directly. some modification is needed.  
I was able to build zeromq.node for target darwin/arm64 successfully. This is what I did:
1. open the file ./script/build.sh and add the line in the cmake command:
```
 -DCMAKE_OSX_ARCHITECTURES="arm64" \
```
2. in the root directory of the repo, run command: `node-gyp rebuild --arch=arm64`
zeromq.node for darwin/arm64 would be generated afterwards.
3. run command to verify if the file is correct:
```
% file build/Release/zeromq.node 
build/Release/zeromq.node: Mach-O 64-bit bundle arm64
```
hope this help
